### PR TITLE
Add functionality for user buying new investments

### LINKED ISSUE #99: Write Python script to fetch single search worth of data

### LINKED ISSUE #100: Write servlet for adding new investment data

### PR DESCRIPTION
**Pull request by roonil-wazlib**
- write python script to fetch contextual data for new investments and update datastore
 - add servlet that adds new investment entries to cloud sql and runs python script if necessary
 - move some functions around for better reusability

Planning on addressing my todo about the timeout tomorrow before merging, but if there are other issues I need to get to then it would be efficient to know before :)

Resolves #100 resolves #99

### CODE REVIEW

File: project/src/main/java/com/google/sps/servlets/NewInvestmentServlet.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+   */
+  private void addUserInvestment(
+      Connection conn, 
+      int userId, 
+      int competitionId, 
+      String googleSearch, 
+      int amtInvested, 
+      Long investDate) throws SQLException {
+    Date date = new Date(investDate * 1000);
+    String stmt = "INSERT INTO investments (user, competition, google_search, invest_date, sell_date, amt_invested) VALUES "
```
Comment (texm) [inline project/src/main/java/com/google/sps/servlets/NewInvestmentServlet.java:110]: would prefer to replace this with a `String.format`

File: project/scripts/main.py [hunk 2]
```diff
@@ -17,6 +17,7 @@
```
Comment (texm) [inline project/scripts/main.py:17]: This file is now no longer the same as what we have in Cloud Functions.
Also we should move the Cloud Functions scripts into a `functions` directory so it's not confusing when the cronjob scripts are created too.

File: project/src/main/java/com/google/sps/servlets/NewInvestmentServlet.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+                ProjectTopicName.of("google.com:sgonks-step272", "trendData")).build();
+        
+            // Attempt to publish the message
+            try {
+                publisher.publish(pubsubApiMessage).get();
+            } catch (InterruptedException | ExecutionException e) {
+                LOGGER.log(Level.SEVERE, "Error publishing Pub/Sub message: " + e.getMessage(), e);
+            }
+        }
+        while (data == null) {
```
Comment (texm) [inline project/src/main/java/com/google/sps/servlets/NewInvestmentServlet.java:86]: maybe wait for .1 seconds between loops? (+timeout)

**Review by texm — APPROVED** [review]
LGTM if the possibly-infinite loop is fixed and comments addressed

File: project/scripts/main.py (hunk 2 above)
Comment (roonil-wazlib) [reply project/scripts/main.py:17]: The cloud functions script and the cronjob script use the exact same functions, the only thing that changes is the entry point. Where would I put the generic files used by both the cronjob and the cloud function? (dates, fetch_trends, database_updates etc). Also, I think I copied the cloud function over exactly into the get_context script, main is the cronjob entry point instead. They're slightly different.

Comment (texm) [reply project/scripts/main.py:17]: Good point. Maybe have the generic files in the `scripts` directory, imported by both `functions` and `cronjobs`?
Would be worth keeping the cloud function entry point in git, as something like `functions/main.py`

**Review by lin8231 — APPROVED** [review]

**Pull request merged**